src/expr/scalar/func.rs

```diff
@@ -425,11 +426,8 @@ fn cast_interval_to_string<'a>(a: Datum<'a>, temp_storage: &'a RowArena) -> Datu
 
 fn cast_interval_to_time<'a>(a: Datum<'a>) -> Datum<'a> {
     let mut i = a.unwrap_interval();
-    if !i.is_positive_dur {
-        i = i + repr::Interval {
-            duration: std::time::Duration::from_secs(86400),
-            ..Default::default()
-        };
+    if i.duration < 0 {
+        i = i + repr::Interval::new(0, 86400, 0);
```

Comment:
This looks like what was here before, but is it easy to explain the logic? It seems like we add one day if the interval is negative, but we might still be negative once that is done. Is that ok?

Reply:
Great catch; thank you. Fixed this and added a test.